src/telliot_core/queries/morphware.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from dataclasses import dataclass
+
+from telliot_core.dtypes.value_type import ValueType
+from telliot_core.queries.abi_query import AbiQuery
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Morphware(AbiQuery):
+    """Returns the result for a given Morphware query version number.
+
+    Attributes:
+        version:
+            A reference to Morphware data specifications found
+            here: www.TODObyMorphware.org/path/to/data/specs
+
+            More about this query:
+            https://github.com/tellor-io/dataSpecs/blob/main/types/Morphware.md
+
+            More about Morphware:
+            https://morphware.org
+    """
+
+    version: int
+
+    #: ABI used for encoding/decoding parameters
+    abi = [{"name": "version", "type": "uint256"}]
+
+    @property
+    def value_type(self) -> ValueType:
+        """Data type returned for a Morphware query.
+
+        - `string`: JSON string containing provider, zones, and instance types
```

Comment:
Hey, the interface for the return type would be like `Array<Ec2MetaData>` where:

```
Interface Ec2MetaData {
   "Instance Type": string.
    "Number of CPUs": string,
    "RAM": string,
    "On-demand Price per Hour": string
}

```
A current endpoint for this would be 
`curl --request POST http://167.172.239.133:5000/products -H "Content-Type: application/json" -d '{"provider":"amazon","service":"compute","region":"us-east-1"}'`

Reply:
> Hey, the interface for the return type would be like `Array<Ec2MetaData>` where:
> 
> ```
> Interface Ec2MetaData {
>    "Instance Type": string.
>     "Number of CPUs": string,
>     "RAM": string,
>     "On-demand Price per Hour": string
> }
> ```
> 
> A current endpoint for this would be `curl --request POST http://167.172.239.133:5000/products -H "Content-Type: application/json" -d '{"provider":"amazon","service":"compute","region":"us-east-1"}'`

Ah, thanks for the clarification!

I'd set the value type that reporters will report to Tellor oracle as a JSON string (before it's converted to bytes and put on chain), because that allows everyone to use the same query type even if y'all want change the specific required data later on (if you add more zones/regions or add different providers). More on that single query type [here](https://github.com/tellor-io/dataSpecs/issues/33).

Using one query type is similar to the setup we've used in our [DIVA Protocol](https://www.divaprotocol.io) integration, where the [`divaProtocolPolygon` query type](https://github.com/tellor-io/dataSpecs/blob/main/types/DivaProtocolPolygon.md) takes one parameter that refers to more specific data requirements on DIVA's end that reporters [can refer to](https://github.com/divaprotocol/oracles/tree/main#diva-queries).

I'm happy to change the reported value type (before converted to bytes) to be `string[]` or any other [ABI type](https://docs.soliditylang.org/en/latest/abi-spec.html#types) you'd like, though using less flexible types may result in us having to keep making new query types if the data requirements change.

Reply:
Hey, thanks for the explanation. I think it makes sense to me (also the specific data requirements you shared for diva gave me a 404).

With regards to the return type, as all of the data will be in a set format, it would make sense to keep it as a string[], where the string is the stringified Ec2MetaData objects